assets/js/modules/analytics/components/common/ProfileSelect.js

```diff
@@ -51,7 +53,7 @@ export default function ProfileSelect() {
 		}
 	}, [ profileID ] );
 
-	if ( isLoadingAccounts || isLoadingProperties || isLoadingProfiles ) {
+	if ( ! hasResolvedAccounts || ( accountID && ! hasResolvedProperties ) || ! hasResolvedProfiles ) {
```

Comment:
Why the `accountID` check in the second part of the if clause? I'd think if we add that, it would make sense to do the same for `accountID` and `propertyID` in the third segment since those are arguments to `getProfiles`.

Reply:
Ok, looks like we don't need it anymore, but in the beginning, I had a situation when profiles select got loaded faster than properties select and having `accountID` check here helped to solve that race condition.

I have removed `accountID`, so it should be good now.

Reply:
Actually, there was an issue when nothing was selected on the setup page. I have updated ProfileSelect and PropertySelect components to check whether fetching is started to prevent progress bars from appearing unexpectedly. Also added a new story to the Analytics setup storybook.